annotation/tests/test_tasks_services.py

```diff
@@ -890,3 +887,144 @@ def test_get_task_revisions(
     assert result[0].validated == [1]
+
+
+def test_get_task_info(mock_session: Mock, create_task: Mock):
+    mock_task = create_task(TaskStatusEnumSchema.ready)
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.first.return_value = mock_task
+    result = services.get_task_info(
+        mock_session, task_id=1, tenant="test_tenant"
+    )
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+    mock_query.first.assert_called_once()
+    assert result == mock_task
+
+
+def test_unblock_validation_tasks(
+    mock_session: Mock,
+    mock_task: ManualAnnotationTask,
+):
+    mock_unblocked_tasks = MagicMock()
+    mock_session.query.return_value.filter.return_value = mock_unblocked_tasks
+    mock_unblocked_tasks.all.return_value = [mock_task]
+    result = services.unblock_validation_tasks(
+        mock_session, mock_task, annotated_file_pages=[1, 2, 3]
+    )
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+    mock_session.query.return_value.filter.assert_called_once()
+    mock_unblocked_tasks.update.assert_called_once_with(
+        {"status": TaskStatusEnumSchema.ready},
+        synchronize_session=False,
+    )
+    assert result == [mock_task]
+
+
+def test_get_task_stats_by_id(mock_session: Mock):
+    mock_stats = AnnotationStatistics()
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value.first.return_value = mock_stats
+    result = services.get_task_stats_by_id(mock_session, task_id=1)
+    mock_session.query.assert_called_once_with(AnnotationStatistics)
+    mock_query.filter.assert_called_once()
+    mock_query.filter.return_value.first.assert_called_once()
+    assert result == mock_stats
+
+
+def test_add_task_stats_record_existing_stats(mock_session: Mock):
+    task_id = 1
+    mock_stats_input = AnnotationStatisticsInputSchema(
+        event_type=AnnotationStatisticsEventEnumSchema.opened
+    )
+    mock_stats_db = AnnotationStatistics()
+    with patch(
+        "annotation.tasks.services.get_task_stats_by_id",
+        return_value=mock_stats_db,
+    ) as mock_get_task_stats_by_id:
+        result = services.add_task_stats_record(
+            mock_session, task_id, mock_stats_input
+        )
+        mock_get_task_stats_by_id.assert_called_once_with(
+            mock_session, task_id
+        )
+        mock_stats_db.field1 = "value1"
+        mock_stats_db.field2 = "value2"
+        mock_stats_db.updated = datetime.utcnow()
+        mock_session.add.assert_called_once_with(mock_stats_db)
+        mock_session.commit.assert_called_once()
+        assert result == mock_stats_db
+
+
+def test_add_task_stats_record(mock_session: Mock):
+    task_id = 1
+    mock_stats_input = AnnotationStatisticsInputSchema(
+        event_type=AnnotationStatisticsEventEnumSchema.closed
+    )
```

Comment:
you dont use this, remove if not needed

Reply:
It is there because I copied it from prev example.